bookshelf/main_test.py

```diff
@@ -0,0 +1,169 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import re
+
+import google.auth
+import main
+import pytest
+import requests
+from six import BytesIO
+
+
+credentials, project_id = google.auth.default()
+os.environ['GOOGLE_CLOUD_PROJECT'] = project_id
+
+
+@pytest.yield_fixture
+def app(request):
+    """This fixtures provides a Flask app instance configured for testing.
+
+    It also ensures the tests run within a request context, allowing
+    any calls to flask.request, flask.current_app, etc. to work."""
+    app = main.app
+
+    with app.test_request_context():
+        yield app
+
+
+@pytest.yield_fixture
+def firestore():
+
+    import firestore
+    """This fixture provides a modified version of the app's firebase model that
```

Comment:
Super-tiny nit: capitalize `firebase` so people know it's a product name?

Also, should this be fire**base** or fire**store**?